index.js

```diff
@@ -1,37 +1,129 @@
+        event.preventDefault();
 
+        const characterName = document.getElementById('searchInput').value;
 
-    })
+        if (characterName) {
+            const url = `https://gateway.marvel.com/v1/public/characters?nameStartsWith=${characterName}&ts=${date}&apikey=${publicKey}&hash=${hash}`;
 
-    .catch(function (error) {
-        console.error('Error fetching data:', error);
+            fetch(url)
+                .then(response => response.json())
+                .then(data => {
+                    console.log(data);
+                    const hero = data.data.results[0];
+
+                    if (hero) {
+                        document.getElementById('heroInfo').innerHTML = `
+                            <img src="${hero.thumbnail.path}.${hero.thumbnail.extension}" alt="${hero.name}">
+                            <h2>${hero.name}</h2>
+                            <p>${hero.description || 'No description available.'}</p>
+                        `;
+                    } else {
+                        document.getElementById('heroInfo').innerHTML = '<p>Character not found.</p>';
+                    }
+                })
+                .catch(error => {
+                    console.error('Error fetching data:', error);
+                    document.getElementById('heroInfo').innerHTML = '<p>Error retrieving data.</p>';
+                });
+        }
     });
 
-    const images = [
-        'image1.jpg',
-        'image2.jpg',
-        'image3.jpg',
-    ]
+    document.getElementById('heroInfo').insertAdjacentHTML('afterend', `
+        <nav>
+            <button id="comicsNavButton">Go to Comics List</button>
+        </nav>`);
 
-    function changeBackground() {
-        const randomImage = images[Math.floor(Math.random() * images.length)];
-        document.body.style.backgroundImage = `url('${randomImage}')`;
-    }
+    document.getElementById('comicsNavButton').addEventListener('click', function () {
+        const characterName = document.getElementById('searchInput').value;
 
-    setInterval(changeBackground, 4000)
+        if (characterName) {
+            const comicsUrl = `https://gateway.marvel.com/v1/public/characters?name=${characterName}&ts=${date}&apikey=${publicKey}&hash=${hash}`;
+
+            fetch(comicsUrl)
+                .then(response => response.json())
+                .then(data => {
+                    const hero = data.data.results[0];
+                    if (hero) {
+                        const characterId = hero.id;
+
+                        fetch(`https://gateway.marvel.com/v1/public/characters/${characterId}/comics?ts=${date}&apikey=${publicKey}&hash=${hash}`)
+                            .then(response => response.json())
+                            .then(comicsData => {
+                                localStorage.setItem('comics', JSON.stringify(comicsData.data.results));
+                                window.location.href = `comics.html?characterId=${characterId}`;
```

Comment:
I think this redirecting you're doing is preventing the page from displaying the character data. That loads the page again and the data is not handled correctly.